### PR TITLE
Getting an email with user profile

### PR DESCRIPTION
**Pull request by cyprianno**
Partial implementation of
https://confluence.atlassian.com/display/BITBUCKET/emails+Resource
with API tree (users->emails) as written in documentation
+ update profile with user's email
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382135%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382157%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382186%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382208%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382255%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382953%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383304%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383343%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383471%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383483%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383484%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383509%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383516%22%2C%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383517%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20ffa89d33d475f6141f47629ac464ea19c23f0fc4%20src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382186%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Seems%20that%20all%20those%20%60%2AOperations%60%20could%20actually%20be%20final.%20That%20would%20require%20to%20inline%20%60initSubApis%28%29%60%20but%20it%20is%20probably%20worth%20it.%20%28Again%2C%20not%20directly%20related%20to%20your%20PR%2C%20feel%20free%20to%20ignore%29.%22%2C%20%22created_at%22%3A%20%222015-06-14T17%3A46%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A46%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java%3AL27-35%22%7D%2C%20%22Pull%2018fccb2248429253188e03c1e397932c2e837f1c%20src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java%2054%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32383509%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20was%20going%20to%20suggest%20to%20add%20a%20comment%20explaining%20your%20intention%2C%20but%20the%20log%20message%20makes%20that%20clear.%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A48%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java%3AL55-91%22%7D%2C%20%22Pull%20ffa89d33d475f6141f47629ac464ea19c23f0fc4%20src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java%2036%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382255%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20there%20a%20reason%20to%20ignore%20other%20HTTP%20errors%3F%20I%20would%20prefer%20to%20let%20the%20exceptions%20bubble%20up.%20The%20client%20of%20that%20API%20probably%20expect%20the%20full%20profile%20to%20be%20loaded%20by%20this%20method.%20At%20least%20we%20should%20log%20a%20warning%20for%20exceptions%20that%20are%20hidden.%22%2C%20%22created_at%22%3A%20%222015-06-14T17%3A51%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22if%20Permissions%3AAcount%3AEmail%20is%20not%20set%2C%20then%20email%20property%20is%20not%20expected%20in%20UserProfile%20by%20client%20%28bitbucket%20will%20answer%20with%20403%29.%20Every%20other%20error%20will%20be%20pushed%20up.%20Warning%20implemented.%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A27%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11454677%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyprianno%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A49%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java%3AL51-74%22%7D%2C%20%22Pull%20ffa89d33d475f6141f47629ac464ea19c23f0fc4%20src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java%2037%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382208%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20would%20add%20a%20static%20import%20for%20%60Arrays.asList%28%29%60.%20%28minor%20comment%2C%20feel%20free%20to%20ignore%29.%22%2C%20%22created_at%22%3A%20%222015-06-14T17%3A47%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22fixed%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A23%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11454677%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyprianno%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A49%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java%3AL1-57%22%7D%2C%20%22Pull%20ffa89d33d475f6141f47629ac464ea19c23f0fc4%20src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java%2043%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382157%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Side%20note%20%28not%20related%20to%20your%20PR%2C%20feel%20free%20to%20ignore%29%3A%20we%20should%20introduce%20Lombok%20for%20generating%20getters%20and%20setters.%22%2C%20%22created_at%22%3A%20%222015-06-14T17%3A44%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A46%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java%3AL1-55%22%7D%2C%20%22Pull%20ffa89d33d475f6141f47629ac464ea19c23f0fc4%20pom.xml%2046%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/gehel/spring-social-bitbucket/pull/12%23discussion_r32382135%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20should%20probably%20go%20to%20a%20profile%2C%20activated%20for%20JDK8.%22%2C%20%22created_at%22%3A%20%222015-06-14T17%3A42%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22fixed%22%2C%20%22created_at%22%3A%20%222015-06-14T18%3A53%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11454677%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cyprianno%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-06-14T19%3A44%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20pom.xml%3AL200-214%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull ffa89d33d475f6141f47629ac464ea19c23f0fc4 pom.xml 46'></a> &nbsp;<b><a href='https://github.com/gehel/spring-social-bitbucket/pull/12#discussion_r32382135'>File: pom.xml:L200-214</a></b>
- <a href='https://github.com/gehel'></a> This should probably go to a profile, activated for JDK8.
- <a href='https://github.com/cyprianno'></a> fixed
- [x] <a href='#crh-comment-Pull ffa89d33d475f6141f47629ac464ea19c23f0fc4 src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java 43'></a> &nbsp;<b><a href='https://github.com/gehel/spring-social-bitbucket/pull/12#discussion_r32382157'>File: src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java:L1-55</a></b>
- <a href='https://github.com/gehel'></a> Side note (not related to your PR, feel free to ignore): we should introduce Lombok for generating getters and setters.
- [x] <a href='#crh-comment-Pull ffa89d33d475f6141f47629ac464ea19c23f0fc4 src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java 16'></a> &nbsp;<b><a href='https://github.com/gehel/spring-social-bitbucket/pull/12#discussion_r32382186'>File: src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java:L27-35</a></b>
- <a href='https://github.com/gehel'></a> Seems that all those `*Operations` could actually be final. That would require to inline `initSubApis()` but it is probably worth it. (Again, not directly related to your PR, feel free to ignore).
- [x] <a href='#crh-comment-Pull ffa89d33d475f6141f47629ac464ea19c23f0fc4 src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java 37'></a> &nbsp;<b><a href='https://github.com/gehel/spring-social-bitbucket/pull/12#discussion_r32382208'>File: src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java:L1-57</a></b>
- <a href='https://github.com/gehel'></a> I would add a static import for `Arrays.asList()`. (minor comment, feel free to ignore).
- <a href='https://github.com/cyprianno'></a> fixed
- [x] <a href='#crh-comment-Pull ffa89d33d475f6141f47629ac464ea19c23f0fc4 src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java 36'></a> &nbsp;<b><a href='https://github.com/gehel/spring-social-bitbucket/pull/12#discussion_r32382255'>File: src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java:L51-74</a></b>
- <a href='https://github.com/gehel'></a> Is there a reason to ignore other HTTP errors? I would prefer to let the exceptions bubble up. The client of that API probably expect the full profile to be loaded by this method. At least we should log a warning for exceptions that are hidden.
- <a href='https://github.com/cyprianno'></a> if Permissions:Acount:Email is not set, then email property is not expected in UserProfile by client (bitbucket will answer with 403). Every other error will be pushed up. Warning implemented.
- [x] <a href='#crh-comment-Pull 18fccb2248429253188e03c1e397932c2e837f1c src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java 54'></a> &nbsp;<b><a href='https://github.com/gehel/spring-social-bitbucket/pull/12#discussion_r32383509'>File: src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java:L55-91</a></b>
- <a href='https://github.com/gehel'></a> I was going to suggest to add a comment explaining your intention, but the log message makes that clear. :+1:

<a href='https://www.codereviewhub.com/gehel/spring-social-bitbucket/pull/12?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/gehel/spring-social-bitbucket/pull/12?approve=1'></a>&nbsp;<a href='https://github.com/gehel/spring-social-bitbucket/pull/12'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -200,10 +200,14 @@
             </plugin>
             <plugin>
                 <artifactId>maven-javadoc-plugin</artifactId>
+                <configuration>
```
Comment (gehel) [inline pom.xml]: This should probably go to a profile, activated for JDK8.

File: src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @JsonProperty("active")
+    private Boolean active;
+
+    @JsonProperty("email")
+    private String email;
+
+    @JsonProperty("primary")
+    private Boolean primary;
+
+    public Boolean getActive() {
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java]: Side note (not related to your PR, feel free to ignore): we should introduce Lombok for generating getters and setters.

File: src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java [hunk 3]
```diff
@@ -30,6 +27,8 @@
 
     private PrivilegeOperations privilegeOperations;
 
+    private UsersOperations usersOperations;
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java]: Seems that all those `*Operations` could actually be final. That would require to inline `initSubApis()` but it is probably worth it. (Again, not directly related to your PR, feel free to ignore).

File: src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ */
+public class EmailsTemplate extends AbstractBitBucketOperations implements EmailsOperations {
+
+    public EmailsTemplate(RestTemplate restTemplate, boolean authorized) {
+        super(restTemplate, authorized, V1);
+    }
+
+    @Override
+    public final List<BitBucketEmailAddress> getListOfUserEmailAddresses(String accountName) {
+        return Arrays.asList(getRestTemplate().getForObject(buildUrl("/users/{accountname}/emails"),
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java]: I would add a static import for `Arrays.asList()`. (minor comment, feel free to ignore).

File: src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java [hunk 5]
```diff
@@ -46,10 +51,23 @@ public final void setConnectionValues(BitBucket api, ConnectionValues values) {
...
+        try {
+            List<BitBucketEmailAddress> emailAddresses = api.usersOperations()
+                    .emailsOperations().getListOfUserEmailAddresses(user.getUsername());
+            for (BitBucketEmailAddress emailAddress : emailAddresses) {
+                if (emailAddress.getPrimary()) {
+                    primaryEmail = emailAddress.getEmail();
+                }
+            }
+        } catch (HttpClientErrorException clientError) {
+            if (!HttpStatus.FORBIDDEN.equals(clientError.getStatusCode())) {
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java]: Is there a reason to ignore other HTTP errors? I would prefer to let the exceptions bubble up. The client of that API probably expect the full profile to be loaded by this method. At least we should log a warning for exceptions that are hidden.

File: pom.xml (hunk 1 above)
Comment (cyprianno) [inline pom.xml]: fixed

File: src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java (hunk 4 above)
Comment (cyprianno) [inline src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java]: fixed

File: src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java (hunk 5 above)
Comment (cyprianno) [inline src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java]: if Permissions:Acount:Email is not set, then email property is not expected in UserProfile by client (bitbucket will answer with 403). Every other error will be pushed up. Warning implemented.

File: pom.xml (hunk 1 above)
Comment (gehel) [inline pom.xml]: :+1:

File: src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java (hunk 2 above)
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java]: :+1:

File: src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java (hunk 3 above)
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/impl/BitBucketTemplate.java]: :+1:

File: src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java [hunk 6]
```diff
@@ -46,13 +55,36 @@ public final void setConnectionValues(BitBucket api, ConnectionValues values) {
...
+                if (emailAddress.getPrimary()) {
+                    return emailAddress.getEmail();
+                }
+            }
+        } catch (HttpClientErrorException clientError) {
+            if (!HttpStatus.FORBIDDEN.equals(clientError.getStatusCode())) {
+                throw clientError;
+            } else {
+                if (logger.isWarnEnabled()) {
+                    logger.warn("BitBucket consumer configuration: Permissions:Acount:Email is not granted."
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java]: I was going to suggest to add a comment explaining your intention, but the log message makes that clear. :+1:

File: src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java (hunk 5 above)
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java]: :+1:<a href='#crh-update-none'></a>

File: src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java (hunk 4 above)
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/impl/EmailsTemplate.java]: :+1:<a href='#crh-update-complete'></a>

File: src/main/java/org/springframework/social/bitbucket/api/EmailsOperations.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+ */
+public interface EmailsOperations {
+
+    /**
+     * Gets the email addresses associated with the account.
+     *
+     * @param accountName The name of an individual or team account.
+     * @return list of email addresses
+     */
+    List<BitBucketEmailAddress> getListOfUserEmailAddresses(String accountName);
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/EmailsOperations.java]: I merged too fast ...

This method should probably be called `getEmailAddressesOfUser()`. List is implied by the return type.

File: src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        return active;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public Boolean getPrimary() {
```
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java]: This should probably by called `isPrimary()` and return a primitive `boolean` (not `Boolean`), unless there is a good reason for it to be `null` (in which case, that reason should be documented in a comment).

**Comment by gehel** [comment]
Thanks for your help! I already merged, but added a few minor comments after the merge. Feel free to have a look at them and send another PR.

Let me know if you want me to create a new release or if the SNAPSHOT is sufficient ...

File: src/main/java/org/springframework/social/bitbucket/api/EmailsOperations.java (hunk 7 above)
Comment (cyprianno) [inline src/main/java/org/springframework/social/bitbucket/api/EmailsOperations.java]: This name refers to the name of command described in docs: https://confluence.atlassian.com/display/BITBUCKET/emails+Resource#emailsResource-GETalistofuser%27semailaddresses (it should be getAListOfUserEmailAddresses)

File: src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java (hunk 8 above)
Comment (cyprianno) [inline src/main/java/org/springframework/social/bitbucket/api/BitBucketEmailAddress.java]: true. will be fixed in next PR.

File: src/main/java/org/springframework/social/bitbucket/api/EmailsOperations.java (hunk 7 above)
Comment (gehel) [inline src/main/java/org/springframework/social/bitbucket/api/EmailsOperations.java]: Ok, let's keep coherence with whatever Atlassian exposes. :+1:

**Comment by cyprianno** [comment]
I'll try to cover the missing parts of BB API (v1 & v2), then it will be ready for 2.0.0 final release.

**Comment by gehel** [comment]
Good ! I'll be waiting !

**Pull request merged**